src/main/target/IRCFUSIONF3/target.h

```diff
@@ -0,0 +1,136 @@
+
+#define USE_MPU_DATA_READY_SIGNAL
+#define ENSURE_MPU_DATA_READY_IS_LOW
+
+#define GYRO
+#define USE_GYRO_MPU6050
+#define GYRO_MPU6050_ALIGN CW270_DEG
+
+#define ACC
+#define USE_ACC_MPU6050
+#define ACC_MPU6050_ALIGN CW270_DEG
+
+#define BARO
+#define USE_BARO_BMP085
+
+#define USE_FLASHFS
+#define USE_FLASH_M25P16
+
+#define USE_UART1
+#define USE_UART2
+#define USE_UART3
+#define USE_SOFTSERIAL1
+#define USE_SOFTSERIAL2
+#define SERIAL_PORT_COUNT 5
+
+#ifndef UART1_GPIO
+#define UART1_TX_PIN        GPIO_Pin_9  // PA9
+#define UART1_RX_PIN        GPIO_Pin_10 // PA10
+#define UART1_GPIO          GPIOA
+#define UART1_GPIO_AF       GPIO_AF_7
+#define UART1_TX_PINSOURCE  GPIO_PinSource9
+#define UART1_RX_PINSOURCE  GPIO_PinSource10
+#endif
+
+#define UART2_TX_PIN        GPIO_Pin_14 // PA14 / SWCLK
+#define UART2_RX_PIN        GPIO_Pin_15 // PA15
+#define UART2_GPIO          GPIOA
+#define UART2_GPIO_AF       GPIO_AF_7
+#define UART2_TX_PINSOURCE  GPIO_PinSource14
+#define UART2_RX_PINSOURCE  GPIO_PinSource15
+
+#ifndef UART3_GPIO
+#define UART3_TX_PIN        GPIO_Pin_10 // PB10 (AF7)
+#define UART3_RX_PIN        GPIO_Pin_11 // PB11 (AF7)
+#define UART3_GPIO_AF       GPIO_AF_7
+#define UART3_GPIO          GPIOB
+#define UART3_TX_PINSOURCE  GPIO_PinSource10
+#define UART3_RX_PINSOURCE  GPIO_PinSource11
+#endif
+
+#define SOFTSERIAL_1_TIMER TIM3
+#define SOFTSERIAL_1_TIMER_RX_HARDWARE 4 // PWM 5
+#define SOFTSERIAL_1_TIMER_TX_HARDWARE 5 // PWM 6
+#define SOFTSERIAL_2_TIMER TIM3
+#define SOFTSERIAL_2_TIMER_RX_HARDWARE 6 // PWM 7
+#define SOFTSERIAL_2_TIMER_TX_HARDWARE 7 // PWM 8
+
+#define USE_I2C
+#define I2C_DEVICE (I2CDEV_1) // PB6/SCL, PB7/SDA
+
+#define USE_SPI
+#define USE_SPI_DEVICE_2 // PB12,13,14,15 on AF5
+
+#define M25P16_CS_GPIO          GPIOB
+#define M25P16_CS_PIN           GPIO_Pin_12
+#define M25P16_SPI_INSTANCE     SPI2
+
+#define USE_ADC
+#define BOARD_HAS_VOLTAGE_DIVIDER
+
+
+#define ADC_INSTANCE                ADC2
+#define ADC_DMA_CHANNEL             DMA2_Channel1
+#define ADC_AHB_PERIPHERAL          RCC_AHBPeriph_DMA2
+
+#define VBAT_ADC_GPIO               GPIOA
+#define VBAT_ADC_GPIO_PIN           GPIO_Pin_4
+#define VBAT_ADC_CHANNEL            ADC_Channel_1
+
+#define CURRENT_METER_ADC_GPIO      GPIOA
+#define CURRENT_METER_ADC_GPIO_PIN  GPIO_Pin_5
+#define CURRENT_METER_ADC_CHANNEL   ADC_Channel_2
+
+#define RSSI_ADC_GPIO               GPIOB
+#define RSSI_ADC_GPIO_PIN           GPIO_Pin_2
+#define RSSI_ADC_CHANNEL            ADC_Channel_12
+
+
+#define BLACKBOX
+#define GPS
+#define GTUNE
+#define SERIAL_RX
+#define USE_SERVOS
+#define TELEMETRY
+#define USE_CLI
+
+#define SPEKTRUM_BIND
+// USART3,
+#define BIND_PORT  GPIOB
+#define BIND_PIN   Pin_11
+
+//#define USE_SERIAL_4WAY_BLHELI_INTERFACE
+#define USE_SERIAL_1WIRE
+#define S1W_TX_GPIO         GPIOA
+#define S1W_TX_PIN          GPIO_Pin_9
+#define S1W_RX_GPIO         GPIOA
+#define S1W_RX_PIN          GPIO_Pin_10
```

Comment:
I will fix #define USE_SERIAL_4WAY_BLHELI_INTERFACE, remove software serial
and make another PR.

On Fri, Apr 22, 2016 at 6:58 AM Yann Oeffner <yann.oeffner@gmail.com> wrote:

> Don't I need software serial somehow for the serial 4 way blheli interface?
>
> On Thu, Apr 21, 2016 at 8:59 PM Dominic Clifton <notifications@github.com>
> wrote:
>
>> In src/main/target/IRCFUSIONF3/target.h
>> <https://github.com/cleanflight/cleanflight/pull/2070#discussion_r60637012>
>> :
>>
>> > +#define SERIAL_RX
>> > +#define USE_SERVOS
>> > +#define TELEMETRY
>> > +#define USE_CLI
>> > +
>> > +#define SPEKTRUM_BIND
>> > +// USART3,
>> > +#define BIND_PORT  GPIOB
>> > +#define BIND_PIN   Pin_11
>> > +
>> > +//#define USE_SERIAL_4WAY_BLHELI_INTERFACE
>> > +#define USE_SERIAL_1WIRE
>> > +#define S1W_TX_GPIO         GPIOA
>> > +#define S1W_TX_PIN          GPIO_Pin_9
>> > +#define S1W_RX_GPIO         GPIOA
>> > +#define S1W_RX_PIN          GPIO_Pin_10
>>
>> use #define USE_SERIAL_4WAY_BLHELI_INTERFACE instead, the old S1W code
>> is gone.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cleanflight/cleanflight/pull/2070/files/9b3494775ece7c1650f0e08c7487c078fa49dc4a#r60637012>
>>
> --
> Meilleures salutations /Herzliche Gruesse/Best regards/最好的祝福送给你
>
>
> www.lemantechlabs.com
>
> Yann Oeffner
>
-- 
Meilleures salutations /Herzliche Gruesse/Best regards/最好的祝福送给你


www.lemantechlabs.com

Yann Oeffner